enterprise/models.py

```diff
@@ -1038,7 +1015,41 @@ def enroll(self, course_run_id, mode, cohort=None, source_slug=None, discount_pe
                     'Error while enrolling user %(user)s: %(message)s',
                     dict(user=self.user_id, message=error_message)
                 )
+
             if succeeded:
+                LOGGER.info(
+                    "[Enroll] LMS enrollment API succeeded for user {username} in course {course_run_id} in "
+                    " mode {mode}".format(
+                        username=self.username,
+                        course_run_id=course_run_id,
+                        mode=mode,
+                    )
+                )
+                try:
+                    enterprise_course_enrollment, created = EnterpriseCourseEnrollment.objects.get_or_create(
```

Comment:
`EnterpriseCourseEnrollment` is only created is course enrollment was successful.